### PR TITLE
[POAE7-2924] Support HDFS in CiderParquetReader

### PR DESCRIPTION
**Pull request by imajiayu**
### What changes were proposed in this pull request?
Support HDFS in CiderParquetReader

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
CiderParquetReaderTest.cpp

### CODE REVIEW

File: cpp/src/cider/util/CiderParquetReader.cpp [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+                  std::string fileName,
+                  std::vector<std::string> requiredColumnNames,
+                  int firstRowGroup,
+                  int rowGroupToRead,
+                  std::string hdfsHost,
+                  int hdfsPort) {
+  if (fsType == LOCAL) {
+    file_ = arrow::io::ReadableFile::Open(fileName).ValueOrDie();
+  } else if (fsType == HDFS) {
+    readHDFS(fileName, hdfsHost, hdfsPort);
```
Comment (jikunshang) [inline cpp/src/cider/util/CiderParquetReader.cpp:66]: initHDFS would be better?

File: cpp/src/cider/util/CiderParquetReader.cpp [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+}
+
+void Reader::init(FileSystemType fsType,
+                  std::string fileName,
+                  std::vector<std::string> requiredColumnNames,
+                  int firstRowGroup,
+                  int rowGroupToRead,
+                  std::string hdfsHost,
+                  int hdfsPort) {
+  if (fsType == LOCAL) {
```
Comment (jikunshang) [inline cpp/src/cider/util/CiderParquetReader.cpp:63]: FileSystemType::Local

File: cpp/src/cider/util/CiderParquetReader.h [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+#include <parquet/api/reader.h>
+#include <string>
+#include <vector>
+#include "exec/plan/parser/TypeUtils.h"
+#include "util/ArrowArrayBuilder.h"
+#include "util/Logger.h"
+
+namespace CiderParquetReader {
+
+enum FileSystemType { LOCAL, HDFS, S3 };
```
Comment (jikunshang) [inline cpp/src/cider/util/CiderParquetReader.h:40]: use enum class instead and use `kLocal, kHdfs, kS3`
https://google.github.io/styleguide/cppguide.html#Enumerator_Names

File: cpp/src/cider/util/CiderParquetReader.cpp [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+    file_ = arrow::io::ReadableFile::Open(fileName).ValueOrDie();
+  } else if (fsType == HDFS) {
+    readHDFS(fileName, hdfsHost, hdfsPort);
+  }
```
Comment (jikunshang) [inline cpp/src/cider/util/CiderParquetReader.cpp:67]: we should also add S3 branch even if we didn't impl it yet.

**Comment by winningsix** [comment]
Close for now and re-open it when needed.

**Pull request closed without merging**